pdns/pdns_recursor.cc

```diff
@@ -1188,9 +1188,16 @@ static void startDoResolve(void *p)
          OPT record.  This MUST also occur when a truncated response (using
          the DNS header's TC bit) is returned."
       */
-      if (addRecordToPacket(pw, makeOpt(edo.d_packetsize, 0, edo.d_Z), minTTL, maxanswersize)) {
```

Comment:
`addRecordToPacket()` takes care of not exceeding the maximum answer size, which the replacement code does not.

Reply:
I was afraid of this indeed